src/Pages/ADT3DScenePage/ADT3DScenePage.tsx

```diff
@@ -231,6 +238,34 @@ const ADT3DScenePageBase: React.FC<IADT3DScenePageProps> = ({
         [deeplinkDispatch, environmentPickerOptions?.adt]
     );
 
+    // if the adx connection information is not in the environment context. fetch it and update the context
+    useEffect(() => {
+        if (!adapter.getADXConnectionInformation()) {
+            {
+                connectionState.callAdapter({
+                    adtInstanceIdentifier: deeplinkState.adtResourceId
+                        ? { id: deeplinkState.adtResourceId }
+                        : { hostName: deeplinkState.adtUrl?.slice(7) }
```

Comment:
how can we avoid copying this "slice(7)" logic all around? can we pull it off to a util function at least with a name? I saw it before in your last PR and this is seems very similar how we got into trouble with the adding and removing https in the environment picker before. 🤔

Reply:
added utils method for those